packages-next/types/src/config/index.ts

```diff
@@ -177,6 +178,13 @@ export type ImagesConfig = {
   };
 };
 
+// config.testing
+
+export type TestingConfig = {
+  isolateRuns?: Boolean;
+  artifactPath?: string;
```

Comment:
When would someone want to use or not use these options?

Reply:
So this is captured in the docs (see other PR), but artifactPath is useful if you want to use a different config to your primary config when running your tests, and isolateRuns is useful if you have more than one config which you want to run tests against. I should include this info in the typescript doc-strings.

Reply:
What would the harm be in having isolateRuns always be true and it not being an option?

> artifactPath is useful if you want to use a different config to your primary config when running your tests

The place that it generates the stuff to defaults to `.keystone/tests` right so how does the config you use when running your tests affect anything?

Reply:
The motivation for `isolateRuns` as an option is that for the vast majority of users, they're going to want this option to be false. Setting it to true for those users would result in tests landing in a slightly unexpected location (`{path}/{hash}`), and will lead to us manipulating their database URL with the schema trick. I'd prefer to avoid having this behaviour unless it's actually necessary, such as in our own tests which use a bunch of different configs.

Good point re: `artifactPath` :-) I had it landing in the primary config location originally, forgot that I'd moved it. We hard-code a bunch of different paths into `.keystone` at the moment, don't we? If that's the case perhaps we can remove this as an option 🤔

Reply:
> tests landing in a slightly unexpected location ({path}/{hash}),

I don't really get why this matters, when will people need to see the files that are generated for their tests? imo we should just write the stuff inside the os temp dir so it gets cleaned up on restarts(and we could make it try to cleanup on process exit as well) which would also mean that artifactPath is unnecessary. And from #5928, it seems like the schema thing is unnecessary so we could also remove isolateRuns and just always generate the stuff into a temp dir(which would be different across prisma schemas) and we wouldn't need to modify their database URL

Reply:
I've done some local testing too and it looks like the schema modification is no longer getting us a win. It looks like the problem this was fixing (paying DB setup overhead when re-running tests locally) has returned in the form of Prisma client setup. A test run now looks something like this:

```
 PASS  tests/api-tests/relationships/crud/one-to-one.test.ts (40.605 s, 136 MB heap size)
  One-to-one relationships
    Read
      ✓ Where A (19182 ms)
      ✓ Where B (797 ms)
...
```
That 15-20s overhead in the first test is coming from the call to `getGenerator`. Ideally this function would recognise if the prisma client had already been generated and act as a no-op, so that subsequent runs of the test didn't have to wait for this to happen, however this doesn't seem to be the case. We should look into this, because fixing it will make running local tests way less painful.

For now I'm going to remove the schema modification stuff.

Reply:
With that gone, it makes setting `isolateRuns: true` as a fixed value less strange, as it doesn't come with the schema hack. Based on the discussion above, this also tips me in favour of hard-coding the location of the generated output to `.keystone/tests/{hash}`, since having a config block with only one option is not very appealing, and we can easily explain this one instance of the hash in the docs.

I'm going to keep the directory in `.keystone` rather than using a temporary directory. When debugging failing tests I find it very helpful to have them living locally, rather than in a system tmp file. I don't think we should be attempting to clean these up for people.